Assembly-CSharp/Preloader.cs

```diff
@@ -31,15 +30,23 @@ Dictionary<string, List<Func<IEnumerator>>> sceneHooks
         var stopwatch = Stopwatch.StartNew();
         MuteAllAudio();
 
+        bool usesSceneHooks = sceneHooks.Sum(kvp => kvp.Value.Count) > 0;
+
         Logger.APILogger.Log($"Preloading using mode {ModHooks.GlobalSettings.PreloadMode}");
         switch (ModHooks.GlobalSettings.PreloadMode) {
-            case "full-scene":
+            case PreloadMode.FullScene:
                 yield return DoPreloadClassic(toPreload, preloadedObjects, sceneHooks, false);
                 break;
-            case "repack-scene":
+            case PreloadMode.RepackScene:
                 yield return DoPreloadClassic(toPreload, preloadedObjects, sceneHooks, true);
                 break;
-            case "repack-assets":
+            case PreloadMode.RepackAssets:
+                if (usesSceneHooks) {
+                    Logger.APILogger.LogWarn($"Some mods ({string.Join(", ", sceneHooks.Keys)}) use scene hooks, falling back to \"RepackScene\" preload mode");
+                    yield return DoPreloadClassic(toPreload, preloadedObjects, sceneHooks, true);
```

Comment:
This name implies to me that it's using FullScene, not RepackScene

Reply:
I changed it to

```cs
IEnumerator DoPreloadAssets(...)
IEnumerator DoPreloadRepackedScenes(...) {
  // repack bundle
  DoPreloadScenes(..., "mapi_preload_assetbundle");
}
IEnumerator DoPreloadScenes(..., sceneprefix = "")

```